docs/hostprovider.md

```diff
@@ -113,31 +117,46 @@ register providers with by calling the `RegisterProvider` method.
 
 #### 2.1.2. Ordering
 
-The default host provider registry in GCM Core will call each host provider in
-the order they were registered in, unless the user has overridden the provider
-selection process.
+The default host provider registry in GCM Core has multiple priority levels that
+host providers can be registered at: High, Normal, and Low.
+
+For each priority level (starting with High, then Normal, then Low), the
+registry will call each host provider in the order they were registered in,
+unless the user has overridden the provider selection process.
 
 There are no rules or restrictions on the ordering of host providers, except
-that the `GenericHostProvider` MUST be registered last. The generic provider is
-a catch-all provider implementation that will handle any request in a standard
-way.
+that the `GenericHostProvider` MUST be registered last and at the Low priority.
+The generic provider is a catch-all provider implementation that will handle any
+request in a standard way.
 
 ### 2.2. Handling Requests
 
-The `IsSupported` method will be called on all registered host providers in-turn
-on the invocation of a `get`, `store`, or `erase` request. The first host
-provider to return `true` will be called upon to handle the specific request.
-If the user has overridden the host provider selection process, a specific host
-provider may be selected instead, and the `IsSupported` method will NOT be
-called.
+The `IsSupported(InputArguments)` method will be called on all registered host
+providers in-turn on the invocation of a `get`, `store`, or `erase` request. The
+first host provider to return `true` will be called upon to handle the specific
+request. If the user has overridden the host provider selection process, a
+specific host provider may be selected instead, and the
+`IsSupported(InputArguments)` method will NOT be called.
 
 This method MUST return `true` if and only if the provider understands the
 request and can serve or handle the request. If the provider does not know how
 to handle the request it MUST return `false` instead.
 
+If no host provider returns `true` to a call to the `IsSupported(InputArguments)`
+method for a each host provider priority level, then a HTTP HEAD request will be
+made to the remote URL and each host provider will be be called via the
+`IsSupported(HttpResponseMessage)` method. A host provider SHOULD use this call
+to check for recognised on-premises instances (for example, by inspecting
+response headers) and return `true` if it wishes to be called upon to handle the
+credential request, otherwise it MUST return `false`.
+
+Host providers SHOULD NOT make further network calls if possible during any of
+the `IsSupported` method overloads to avoid degrading the performance of the
+overall application.
+
 #### 2.2.1. Rejecting Requests
 
-The `IsSupported` method MUST return `true` if the host provider would like to
+The `IsSupported` methods MUST return `true` if the host provider would like to
```

Comment:
Are there multiple overloads of `IsSupported` for a single extension to implement?

Reply:
Yes. `IsSupported(InputArguments)` and `IsSupported(HttpResponseMessage)`.